Initial version of Adventure Engine 2.0.0

danigm opened the pull request.

Barthalion commented:
bot, build com.bladecoder.adventure-editor

Barthalion commented:
You need to run `autoreconf -fi` for arm and aarch64 or copy updated config.sub from `/usr/share/gnu-config/config.sub`.

danigm commented:
I've updated the repository to work with aarch64 as @Barthalion suggested, adding `autoreconf -if`. The arm version of the org.freedestkop.Sdk.Extension.openjdk9 is broken right now, so that will fail. How can I remove the arm arch meantime is fixed? with the `only-arches` in all sources? or there's a global `arches` setting?

Barthalion commented:
You can create a `flathub.json` file with content like this:

```
{
  "skip-arches": [ "arm" ]
}
```

danigm commented:
This last version should compile, I've removed the problematic arm arch

nedrichards commented:
bot, build com.bladecoder.adventure-editor

nedrichards commented on com.bladecoder.adventure-editor.appdata.xml:
```diff
@@ -0,0 +1,22 @@
+  <summary>Classic point and click adventure game engine and editor</summary>
+  <url type="homepage">https://github.com/bladecoder/bladecoder-adventure-engine</url>
+  <description>
+    The Bladecoder Adventure Engine is a set of tools to create interactive graphic adventures (classical point and click games).
```
This needs to be wrapped in <p> tags

nedrichards commented:
This should have an [OARS profile](https://hughsie.github.io/oars/) in the appdata.

nedrichards commented on com.bladecoder.adventure-editor.json:
```diff
@@ -0,0 +1,79 @@
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
```
Does this actually work? I don't know if Java has wayland support out of the box.

danigm replied:
It works in wayland, I don't know if it's using the X11 compatibility layer

TingPing replied:
> It works in wayland, I don't know if it's using the X11 compatibility layer

 It surely doesn't. In GNOME-Shell you can hit alt+f2, run `lg`, then in windows click on one to see if it is an X11Window.

danigm replied:
![imagen](https://user-images.githubusercontent.com/50350/36300764-bcdfe27c-1302-11e8-8f14-b8c51f786f25.png)

All my windows have the GType:MetaWindowX11.

barthalion replied:
So it doesn't support Wayland. Can you remove it? We will give it last test build and merge.

danigm replied:
Done

nedrichards commented:
@danigm It'd be great to get the desktop file, appdata, icon etc. upstream.

danigm commented:
@nedrichards, I'll make a pull request for the desktop, appdata and icon in upstream.

The OARS is a new requirements? It's not a game, it's a dev tool to make games. I've not add any OARS information in my other flatpak.

nedrichards commented:
If you look at the new OARS site there's a flow for 'non game apps'. https://hughsie.github.io/oars/

OARS is in the [app requirements](https://github.com/flathub/flathub/wiki/App-Requirements) now but we're still being a bit flexible in transition, since very few apps have it yet.

danigm commented:
I've added OARS. I've also added a pull request in upstream, but the current release doesn't have it so this version is the correct one for the 2.0.0 version.

https://github.com/bladecoder/bladecoder-adventure-engine/pull/35

For the next release, I'll update the flatpak to use the upstream appdata, .desktop and icons

nedrichards commented:
Thanks!

nedrichards commented:
bot, build com.bladecoder.adventure-editor

Barthalion commented:
bot, build com.bladecoder.adventure-editor

nedrichards commented:
bot, build com.bladecoder.adventure-editor

TingPing commented on com.bladecoder.adventure-editor.json:
```diff
@@ -0,0 +1,84 @@
+    "modules" : [
+        {
+            "name": "xorg-xrandr",
+            "buildsystem": "simple",
```
This should be the regular build system, you aren't doing anything advanced on top. You can make an `autogen.sh` with the `script` type source file that calls `autoreconf -fi` if needed.

danigm replied:
Sorry, I don't understand what should I do here. Can you point me to other project that do this or to the doc to view how to change this?

TingPing replied:
Like this: https://github.com/flathub/org.DolphinEmu.dolphin-emu/blob/98d37a870252a19e8e31e0b3841eb24788904526/org.DolphinEmu.dolphin-emu.json#L49-L55

danigm replied:
I'm trying adding this to the xorg sources, but doesn't work for me:

```
                {
                     "type": "script",
                     "dest-filename": "autogen.sh",
                     "commands": [
                          "autoreconf -fi"
                     ]
                }
```

I think that the autogen.sh isn't called. If I use type `shell` instead of `script` it works.

TingPing commented on com.bladecoder.adventure-editor.json:
```diff
@@ -0,0 +1,84 @@
+            "name": "adventure-editor",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -d /app/bin",
```
You can combine these in the other install commands with `-D`

nedrichards commented:
bot, build com.bladecoder.adventure-editor

danigm commented:
What do you think about the last version of the package? it's okay for flathub?

barthalion commented:
bot, build com.bladecoder.adventure-editor

nedrichards commented:
Repository has been created: https://github.com/flathub/com.bladecoder.adventure-editor
